### PR TITLE
feat: workspace details/project list page [DET-6825]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Merging into the Workspaces feature branch

Screenshot 2022-04-15 at 15-39-45 Determined
Screenshot 2022-04-15 at 15-39-24 Determined

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/pages/WorkspaceDetails/ProjectCard.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            {project.name}
+          </Link>
+        </h6>
+        {!project.immutable && (
+          <ProjectActionDropdown className={css.action} curUser={curUser} project={project} />
+        )}
+        <p className={css.description}>{project.description}</p>
+        <div className={css.experiments}>
+          <Icon name="experiment" size="small" />
+          <span>{project.numExperiments}</span>
```
Comment (mapmeld) [inline webui/react/src/pages/WorkspaceDetails/ProjectCard.tsx:34]: Screen Shot 2022-04-15 at 3 29 11 PM
nit picks: Could this be `.toLocaleString()` (for >1,000 experiments) and some margin or padding

File: webui/react/src/pages/WorkspaceDetails.tsx [hunk 2]
```diff
@@ -1,28 +1,324 @@
...
+      id="workspaceDetails">
+      <div className={css.controls}>
+        <SelectFilter
+          bordered={false}
+          label="View:"
+          value={projectFilter}
+          onSelect={handleViewSelect}>
+          <Option value={ProjectFilters.All}>All projects</Option>
+          <Option value={ProjectFilters.Mine}>My projects</Option>
+          <Option value={ProjectFilters.Others}>Others projects</Option>
```
Comment (mapmeld) [inline webui/react/src/pages/WorkspaceDetails.tsx:292]: nit pick: I would add the apostrophe: Others'

**Comment by mapmeld** [comment]
I can't make a comment directly on the Move Project modal, but how do you want to handle the Uncategorized workspace since it can't add or receive new projects.  Either there could be an error message in the future, or adding `.filter(w => !w.immutable)` so we don't see that workspace in the list.

**Comment by mapmeld** [comment]
When I select "Others Projects" the Network tab shows it is making queries over and over again, multiple times per second. Most of this code looks standard so I don't know what causes that.

File: webui/react/src/pages/WorkspaceDetails/ProjectCard.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/pages/WorkspaceDetails/ProjectCard.tsx]: Screen Shot 2022-04-18 at 10 29 56 AM

I still have an issue with no whitespace between the number and the time, it looks like the position is set by the grid and any more digits would make the count and time overlap.

Comment (EmilyBonar) [reply webui/react/src/pages/WorkspaceDetails/ProjectCard.tsx]: Alright, I fixed this for real this time.

**Comment by mapmeld** [comment]
I have a question about 'Archive' -- it looks like by default the query has archived=false, so archiving a project hides it from the UI.
I don't know if this would be better in a future PR (?) but I don't have a way to remove the archived projects filter from the list.

**Review by mapmeld — APPROVED** [review]
Approved with the note about archived projects are not able to be un-filtered

**Pull request merged**